packages/webapp/pages/devcard.tsx

```diff
@@ -187,10 +187,15 @@ const Step2 = ({ initialDevCardSrc }: Step2Props): ReactElement => {
 
   const onUpdatePreference = useCallback(
     (props: Partial<Omit<GenerateDevCardParams, 'type'>>) => {
-      client.setQueryData(key, (oldData: DevCardData) => ({
-        ...oldData,
-        ...props,
-      }));
+      client.setQueryData(key, (oldData: DevCardQueryData) => {
+        return {
+          ...oldData,
+          devCard: {
+            ...oldData.devCard,
```

Comment:
Maybe just optional `oldData?.devCard` handler if `oldData` is undefined for some reason